### ISSUE TITLE
Google authentication not working with Expo SDK 48

### ISSUE THREAD

**Issue by Rashed94x** [expo/expo#21944]
### Minimal reproducible example

https://github.com/Rashed94x/expo48-google-auth

### Summary

Expected behavior

When the user clicks on the sign-in button, a pop-up web browser should open, allowing the user to select a Google account to use for logging in to the Expo app. The user should then be redirected back to the app and authenticated with their selected Google account.

Actual behavior

After upgrading to Expo SDK 48 and attempting to sign in with Google, a web browser opens with the error message "ValidationError: 'client_id' is not allowed. 'response_type' is not allowed. 'state' is not allowed. 'scope' is not allowed." The user is not able to select a Google account and authenticate with the app.

Affected platforms

This issue affects iOS devices when using Expo SDK 48.

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
      npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
    npmPackages:
      expo: ~48.0.9 => 48.0.9 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.4 => 0.71.4 
    Expo Workflow: managed

**Comment by truongtv22** [comment]
Same issue =((

**Comment by RaymondMoay** [comment]
Same issue. Using Firebase JS SDK.

**Comment by SirRamboia** [comment]
Same issue. 😔

**Comment by timsmith339** [comment]
Has anybody found a possible workaround?

**Comment by RaymondMoay** [comment]
Works after a rebuild. Thanks!

**Comment by alexwert10** [comment]
Has anyone been able to get this working for a preview build? Your comments helped me get the development build working with Google Auth through firebase, but when I switch to a preview build II get the following error: "Access blocked: This app's request is invalid. You can't sign in because this app sent an invalid request. You can try again later, or contact the developer about this issue. Error 400: redirect_uri_mismatch".

**Comment by JDoveBizDev** [comment]
Ya same issue. Would rather not use firebase auth because we're using mongo db. Getting the validation error on iOS

**Comment by karimcambridge** [comment]
It appears that expo is slowly forcing everybody to use dev clients.

add `useProxy: true` in `useAuthRequest`, it will give you a deprecation warning but it will work in expo go.

for expo-dev-client, use `makeRedirect`, follow the instructions in the docs.

**Comment by amandeepmittal** [comment]
@Rashed94x, (going through the minimal reproducible example link) Instead of `expoClientId`, please use `androidClientId` and `iosClientId` for native platforms. We do mention that in docs: https://docs.expo.dev/guides/google-authentication/#get-credentials. Also, when getting Client IDs from Google Cloud, make sure you are generating separate ids for native platforms and selecting native application type (instead of Web which was the case previously when using Expo Go but with development builds, you have to select the native application).

**Comment by Rashed94x** [comment]
> Instead

Hey @amandeepmittal 
To workaround this problem in the project i was working on (which was a school project) i just downgraded to EXPO SDK 47.
However, your response has prompted a new question for me.

Are you suggesting that to use the AuthSession API with iOS on Expo SDK 48, it's necessary to utilize development builds? If that's the case, does that means that i have to work on a macOS i order to run an ios simulator to run the development build?

in other words, is it necessary to use macOS to (develop and test) AuthSession API with ios?

**Comment by Rashed94x** [comment]
> @Rashed94x, (going through the minimal reproducible example link) Instead of `expoClientId`, please use `androidClientId` and `iosClientId` for native platforms. We do mention that in docs: https://docs.expo.dev/guides/google-authentication/#get-credentials. Also, when getting Client IDs from Google Cloud, make sure you are generating separate ids for native platforms and selecting native application type (instead of Web which was the case previously when using Expo Go but with development builds, you have to select the native application type).

Hey @amandeepmittal 
To workaround this problem in the project i was working on (which was a school project) i just downgraded to EXPO SDK 47.
However, your response has prompted a new question for me.

Are you suggesting that to use the AuthSession API with iOS on Expo SDK 48, it's necessary to utilize development builds? If that's the case, does that means that i have to work on a macOS i order to run an ios simulator to run the development build?

in other words, is it necessary to use macOS to (develop and test) AuthSession API with ios?

**Comment by amandeepmittal** [comment]
Yes. We deprecated authProxy from SDK 48 (more info in [this blog post](https://blog.expo.dev/security-advisory-for-developers-using-authsessions-useproxy-options-and-auth-expo-io-e470fe9346df) as to _why_) and moving forward, we recommend creating development builds because `androidClientId` and `iosClientId` are both native client IDs whereas `expoClientId` used to be a Proxy that was meant to be used for testing purposes. 

You can use [EAS to create a development build on the cloud](https://docs.expo.dev/develop/development-builds/create-a-build/) and use an iOS device to test it or if you have access to macOS with Xcode installed, you can [create development builds locally](https://docs.expo.dev/develop/development-builds/development-workflows/#build-locally-with-android-studio-and-xcode) and use iOS simulator for testing.

**Comment by amandeepmittal** [comment]
We've deprecated Google auth provider from `expo-auth-session` starting SDK 49. We recommend using a native library such as [@react-native-google-signin/google-signin](https://github.com/react-native-google-signin/google-signin). We don't maintain the native library, however, we did write a short guide on what steps might be required to configure an Expo project: https://docs.expo.dev/guides/google-authentication/. The same library can also be used with previous SDK versions.

**Comment by ansh** [comment]
@amandeepmittal If I want to implement Google sign in on an Expo Go application, is it no longer possible?

**Comment by amandeepmittal** [comment]
@ansh, with SDK 49, it is not. You'll have to create a development build and use a library like  https://github.com/react-native-google-signin/google-signin since it modifies custom native code via a config plugin.

**Comment by ansh** [comment]
@amandeepmittal That is unfortunate! The package you linked to requires payment for most new functionality and even bug fixes.

**Comment by magnum6actual** [comment]
It seems like there is less and less of a reason to even use Expo going forward. Is the entire project being shut down?

**Comment by sonipranjal** [comment]
This is really not a great thing, I spent almost a day trying to implement google oauth for react native, and here I see it is not possible with SDK 49, and this package https://github.com/react-native-google-signin/google-signin doesn't support supabase well, what options are left?

**Comment by MTahaSh** [comment]
Can anyone provide a complete guide on how to implement it accurately in expo go?
